diff --git a/include/config/ftoption.h b/include/freetype/ftoption.h
index 1536c72..c64c116 100644
--- a/include/config/ftoption.h
+++ b/include/config/ftoption.h
@@ -279,6 +279,13 @@ FT_BEGIN_HEADER
   /*   Do not #undef these macros here since the build system might define */
   /*   them for certain configurations only.                               */
   /*                                                                       */
+#ifdef freetype_EXPORTS
+#define FT_EXPORT(x) __declspec(dllexport) x
+#define FT_EXPORT_DEF(x)  __declspec(dllexport) x
+#else
+#define FT_EXPORT(x) __declspec(dllimport) x
+#define FT_EXPORT_DEF(x)  __declspec(dllimport) x
+#endif
 /* #define FT_EXPORT(x)      extern x */
 /* #define FT_EXPORT_DEF(x)  x */
 
